sys/net/gnrc/priority_pktqueue/priority_pktqueue.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @defgroup    net_gnrc_priority_pktqueue Priority packet queue
+ * @ingroup     net_gnrc
+ * @file
+ * @brief       Wrapper for priority_queue that holds gnrc_pktsnip_t* and is
+ *              aware of it's length.
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo <shuguo.zhuo@inria.fr>
+ * @}
+ */
+
+#include "net/gnrc/pktbuf.h"
+#include <net/gnrc/priority_pktqueue.h>
+
+/******************************************************************************/
+
+static inline void _free_node(gnrc_priority_pktqueue_node_t *node)
+{
+    assert(node != NULL);
+
+    priority_queue_node_init((priority_queue_node_t *)node);
+}
+
+/******************************************************************************/
+
+gnrc_pktsnip_t* priority_pktqueue_pop(gnrc_priority_pktqueue_t* q)
+{
+    if(!q || (priority_pktqueue_length(q) == 0))
```

Comment:
[Style](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#indentation-and-braces)